distributed/tests/test_worker.py

```diff
@@ -1789,6 +1796,36 @@ async def test_story(c, s, w):
     assert w.story(ts) == w.story(ts.key)
 
 
+@gen_cluster(client=True)
+async def test_story_with_deps(c, s, a, b):
+    """
+    Assert that the structure of the story does not change unintentionally and
+    expected subfields are actually filled
+    """
+    future = c.map(inc, range(10))
+    res = c.submit(sum, future)
+    await res
+
+    for w in [a, b]:
+        try:
+            ts = w.tasks[res.key]
+        except KeyError:
+            continue
+        story = w.story(ts)
+        for msg in story:
+            # There are multiple types of messages
+            # Transition
+            if isinstance(msg[-1], str):
+                assert msg[0] == ts.key
+            # events, e.g. gather-deps
+            elif msg[0] == "gather-dependencies":
```

Comment:
We might also want to `assert` earlier on that that at least one of the messages is a `gather-dependencies` message so we know for sure we go into this branch